### ISSUE TITLE
'vat' doesn't work as expected

### ISSUE THREAD

**Issue by geg-mopo3** [vim/vim#3927]
I've attached a zip file that contains an xml file that can be used to reproduce the issue.

Steps to reproduce:
1) Open the xml in vim
2) Go to line 42
3) Run vat in normal mode to select an XML element under cursor
4) Note that the lines 42 to 141 are selected although it should be the lines 42 to 199

[source.zip](https://github.com/vim/vim/files/2848110/source.zip)

Reproduced in 
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Apr 10 2018 21:31:58)
Included patches: 1-1453

**Comment by tom-m** [comment]
I am able to reproduce this issue with debian's Vim 8.0 (Included patches: 1-197, 322, 377-378, 550, 703, 706-707). Experimenting with the given example, I've come to a tiny xml that possibly manifests the same behaviour. Use `vat` on the first line of this xml:
```xml
<two>
  <two
    id="abc">
  </two>
</two>
```
And now compare with the same op on a slightly modified xml:
```xml
<two>
  <two id="abc">
  </two>
</two>
```
Seems like having the tag's attributes on the same line as the attribute itself helps to mitigate this. Similar modification of OP's xml file - joining lines 49-55 and 143-148 - can be used as a workaround for his/her case as well.

**Comment by tom-m** [comment]
FWIW, same behaviour wih Vim 8.1.526.

Tom

**Comment by tonymec** [comment]
I confirm the problem on Vim 8.1.883, the latest so far except for a couple of changes affecting only tests.

A workaround (with matchit enabled) would consist of selecting the starting tag, here: linewise, with V — then placing the cursor at least one character after the opening < and hitting %

This is of course a workaround, not a fix.

Best regards,
Tony.

**Comment by lacygoill** [comment]
Fixed by [8.2.1850](https://github.com/vim/vim/releases/tag/v8.2.1850).
